app/src/main/java/org/mifos/mobile/ui/recent_transactions/RecentTransactionScreen.kt

```diff
@@ -0,0 +1,275 @@
+    navigateBack: () -> Unit )
+{
+    val uiState by viewModel.recentTransactionUiState.collectAsStateWithLifecycle()
+    val isRefreshing by viewModel.isRefreshing.collectAsStateWithLifecycle()
+
+    LaunchedEffect(key1 = Unit) {
+        viewModel.loadRecentTransactions(false, 0)
+    }
+
+   RecentTransactionScreenContent(
+        uiState = uiState,
+        navigateBack = navigateBack,
+        retryConnection = { viewModel.loadRecentTransactions(false, 0) },
+        isRefreshing = isRefreshing,
+        onRefresh = { viewModel.refresh() },
+        viewModel= viewModel
+    )
+}
+
+@OptIn( ExperimentalMaterial3Api::class)
+@Composable
+fun RecentTransactionScreenContent(
+    uiState: RecentTransactionUiState,
+    navigateBack: () -> Unit,
+    retryConnection: () -> Unit,
+    isRefreshing: Boolean,
+    onRefresh: () -> Unit,
+    viewModel: RecentTransactionViewModel
+) {
+
+    val context = LocalContext.current
+    val pullRefreshState = rememberPullToRefreshState()
+    val scrollState = rememberScrollState()
+    val lazyColumnState = rememberLazyListState()
+
+        Box(
+            Modifier
+                .fillMaxSize()
+                .nestedScroll(pullRefreshState.nestedScrollConnection)){
+            Column(modifier = Modifier
+                .fillMaxSize()
+                .verticalScroll(scrollState), verticalArrangement = Arrangement.Center) {
+
+                when (uiState) {
+
+                    RecentTransactionUiState.EmptyTransaction -> {
+                        MifosErrorComponent( isEmptyData = true)
+                    }
+
+                    is RecentTransactionUiState.Error -> {
+                        MifosErrorComponent(
+                            isNetworkConnected = Network.isConnected(context),
+                            isEmptyData = false,
+                            isRetryEnabled = true,
+                            onRetry = retryConnection
+                        )
+                    }
+
+                    RecentTransactionUiState.Initial -> {
+
+                    }
+
+                    RecentTransactionUiState.Loading -> {
+
+                        MifosProgressIndicator(
+                            modifier = Modifier
+                                .fillMaxSize()
+                                .padding(10.dp)
+                                .background(MaterialTheme.colorScheme.background.copy(alpha = 0.7f))
+                        )
+                    }
+
+                    is RecentTransactionUiState.LoadMoreRecentTransactions -> {
+                        LazyColumn(
+                            modifier = Modifier.fillMaxSize(),
+                            state = lazyColumnState
+                        ) {
+                            items(uiState.transactions) { transaction ->
+                                RecentTransactionListItem(transaction)
+                            }
+                        }
```

Comment:
i dont think we need to another compose UI for this . The idea is to update the data source that is fetching all the recent transactions